Первая версия(SVD)

MaEgV opened the pull request:
Умеет строить частотную матрицу по списку предложений, разбитому по словам
Сингулярное разложение матрицы и срезы первых к строк и столбов необходимых матриц
Проблемы:
  Неоптимальный механизм соответствия номера строки и слова
  Лишняя пустая строка в матрице
Нужно:
  Доделать соответствие слов предложениям на финальном этапе и продумать вывод результатов
  Фикс проблем

MaEgV committed:
Первая версия

Умеет строить частотную матрицу по списку предложений, разбитому по словам
Сингулярное разложение матрицы и срезы первых к строк и столбов необходимых матриц
Проблемы:
1) Неоптимальный механизм соответствия номера строки и слова
2) Лишняя пустая строка в матрице
Нужно:
1) Доделать соответствие слов предложениям на финальном этапе и продумать вывод результатов
2) Фикс проблем

proted reviewed (changes requested):
1) К _**каждой**_ функции, классу и всему прочему (в том числе и фрагментам кода) оставить развернутый комментарий, поясняющий входы-выходы реализуемого алгоритма, его принцип работы и (опциональо, но желательно) обоснованность использования тех или иных структур данных или чего-то еще
2) Весь функциональный код должен быть завернут в функции, чтобы это все можно было подключить как модуль в основном файле нашей ИС

MaEgV committed:
Документация функций

Откомментил существующие функции

MaEgV committed:
Update определения тематики

Старый функционал приобрёл чуть более адекватную реализацию
Новый функционал:
Всё доведено до момента, когда можно выбирать из списка тем те, которые мы хотим (например 3 самых важных в разговоре)
В 1.45 мне кажется, что это логично и должно работать, если я утром 30.04 не удалю этот коммит, значит всё идет по плану
Из плюсов:
Оно интерпретируется без ошибок
Из минусов:
Тестов пока не было, но с помощью модуля Лизы Гришиной мы сможем их провести
На будущее:
Есть варианты немного повысить точность, если это будет необходимо (строка 49)
Возможно! есть варианты сильно выиграть по памяти и мб по времени (более изящное решение с парами (слово-номер строки))

abyzgareeva commented:
Что будет, если тема не определяется, то есть вспомогательные слова не встречаются?

MaEgV commented:
Пока будет возвращен либо пустой список, либо список с нулевыми весами каких-то тем
Это не готовый продукт, я пока оставляю код, удобный для разработки
В будущем, будет возвращен список тем из одной темы "другое"

MaEgV committed:
Версия для интеграции

Функция topics ВОЗРАЩАЕТ список тем
